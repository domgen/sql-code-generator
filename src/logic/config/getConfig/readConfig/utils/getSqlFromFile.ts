import { readFileAsync } from '../../../_utils/readFileAsync';

export const getSqlFromFile = async ({ filePath }: { filePath: string }): Promise<string> => {
  const extension = filePath.split('.').slice(-1)[0];
  if (extension === 'sql') return readFileAsync({ filePath });
  if (extension === 'ts') {
    const exports = await import(filePath);
    if (!exports.sql) throw new Error('.ts file did not have an export with name "sql"');
    return exports.sql;
  }
  throw new Error('path must lead to file with extension of .ts or .sql');
};
