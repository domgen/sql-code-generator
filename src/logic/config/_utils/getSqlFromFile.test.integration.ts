import { getSqlFromFile } from './getSqlFromFile';

describe('getSqlFromFile', () => {
  it('should be able to read sql from a sql file', async () => {
    const contents = await getSqlFromFile({ filePath: `${__dirname}/__test_assets__/user.table.sql` });
    expect(contents).toMatchSnapshot();
  });
  it('should be able to read sql from a ts file with a named "sql" export', async () => {
    const contents = await getSqlFromFile({ filePath: `${__dirname}/__test_assets__/findUserById.ts` });
    expect(contents).toMatchSnapshot();
  });
});
