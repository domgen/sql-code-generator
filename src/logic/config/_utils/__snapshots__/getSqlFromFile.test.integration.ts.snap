// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSqlFromFile should be able to read sql from a sql file 1`] = `
"CREATE TABLE user (
  id BIGINT,
  name VARCHAR(255)
)
"
`;

exports[`getSqlFromFile should be able to read sql from a ts file with a named "sql" export 1`] = `"SELECT * FROM user WHERE id = :id"`;
